### PR TITLE
code enhancement in `sort.double_sort`

### PR DESCRIPTION
**Pull request by Krishna-Singhal**
### Describe your change:
- Don't need to return list because list is mutable
- use advantage of python in swapping
- filter blank inputs from input list
- minor changes

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Comment by Krishna-Singhal** [comment]
It would help me if you could add label `hacktoberfest-accepted`, Thanks...

**Comment by cclauss** [comment]
Let's change line 1 to a commitment to return a tuple:
```python
def double_sort(unsorted: list) -> tuple:
```
Let's change the vague name `lst` to `unsorted` which is more self-documenting.

**Comment by cclauss** [comment]
No label until the tests pass and it is merged.  ;-)

As discussed in `CONTRIBUTING.md`, you can run the `build` tests on your computer with the commend:
```
python3 -m doctest -v sorts/double_sort.py
```

**Comment by Krishna-Singhal** [comment]
> As discussed in `CONTRIBUTING.md`, you can run the `build` tests on your computer with the command:
> 
> ```
> python3 -m doctest -v sorts/double_sort.py
> ```

All tests are passed on my computer.

**Comment by cclauss** [comment]
The functions returns `None` so the failure on line 807 makes sense to me.
https://github.com/TheAlgorithms/Python/actions/runs/6602527016/job/17934662337?pr=10801#step:6:808

**Comment by Krishna-Singhal** [comment]
> The functions returns `None` so the failure on line 807 makes sense to me. https://github.com/TheAlgorithms/Python/actions/runs/6602527016/job/17934662337?pr=10801#step:6:808

could you review it please?

File: sorts/double_sort.py [hunk 1]
```diff
@@ -14,29 +17,32 @@ def double_sort(lst):
...
+                collection[j + 1] < collection[j]
             ):  # applying bubble sort algorithm from left to right (or forwards)
-                temp = lst[j + 1]
-                lst[j + 1] = lst[j]
-                lst[j] = temp
+                collection[j], collection[j + 1] = collection[j + 1], collection[j]
             if (
-                lst[no_of_elements - 1 - j] < lst[no_of_elements - 2 - j]
+                collection[no_of_elements - 1 - j] < collection[no_of_elements - 2 - j]
             ):  # applying bubble sort algorithm from right to left (or backwards)
```
Comment (cclauss) [inline sorts/double_sort.py:31]: Let's not wrap lines of code just to make room for a trailing comment.  Please put the comment on a line of its own _before_ the `if` statement.  Please repeat on any other comments in this file that force a line to be wrapped.
```suggestion
            ):  # applying bubble sort algorithm from right to left (or backwards)
```

File: sorts/double_sort.py [hunk 2]
```diff
@@ -14,29 +17,32 @@ def double_sort(lst):
...
 
 
 if __name__ == "__main__":
-    print("enter the list to be sorted")
-    lst = [int(x) for x in input().split()]  # inputing elements of the list in one line
-    sorted_lst = double_sort(lst)
+    print("Enter the list to be sorted: ")
+    unsorted = [
+        int(x) for x in input().split() if x
+    ]  # inputing elements of the list in one line
```
Comment (cclauss) [inline sorts/double_sort.py:46]: Let's not wrap lines of code just to make room for a trailing comment.

Comment (cclauss) [inline sorts/double_sort.py:46]: Let's not wrap lines of code just to make room for a trailing comment.
```suggestion
    # allow the user to input the elements of the list on one line
    unsorted = [int(x) for x in input("Enter the list to be sorted: ").split() if x]
```

File: sorts/double_sort.py [hunk 3]
```diff
@@ -14,29 +17,32 @@ def double_sort(lst):
...
     print("the sorted list is")
-    print(sorted_lst)
+    print(double_sort(unsorted))
```
Comment (cclauss) [inline sorts/double_sort.py:48]:
```suggestion
    print(f"{double_sort(unsorted) = }")
```

File: sorts/double_sort.py [hunk 4]
```diff
@@ -14,29 +17,32 @@ def double_sort(lst):
...
     ):  # we don't need to traverse to end of list as
         for j in range(no_of_elements - 1):
             if (
-                lst[j + 1] < lst[j]
+                collection[j + 1] < collection[j]
             ):  # applying bubble sort algorithm from left to right (or forwards)
-                temp = lst[j + 1]
-                lst[j + 1] = lst[j]
-                lst[j] = temp
+                collection[j], collection[j + 1] = collection[j + 1], collection[j]
```
Comment (cclauss) [inline sorts/double_sort.py:28]:
```suggestion
                collection[j], collection[j + 1] = collection[j + 1], collection[j]
              # apply the bubble sort algorithm from right to left (or backwards)

```

File: sorts/double_sort.py [hunk 5]
```diff
@@ -14,29 +17,32 @@ def double_sort(lst):
...
             if (
```
Comment (cclauss) [inline sorts/double_sort.py:29]:
```suggestion
            # apply the bubble sort algorithm from right to left (or backwards)
            if (
```

File: sorts/double_sort.py [hunk 6]
```diff
@@ -14,29 +17,32 @@ def double_sort(lst):
     >>> double_sort([-3, 10, 16, -42, 29]) == sorted([-3, 10, 16, -42, 29])
     True
     """
-    no_of_elements = len(lst)
+    no_of_elements = len(collection)
     for _ in range(
         int(((no_of_elements - 1) / 2) + 1)
     ):  # we don't need to traverse to end of list as
         for j in range(no_of_elements - 1):
             if (
```
Comment (cclauss) [inline sorts/double_sort.py:25]:
```suggestion
            # apply the bubble sort algorithm from left to right (or forwards)
            if (
```

File: sorts/double_sort.py [hunk 7]
```diff
@@ -14,29 +17,32 @@ def double_sort(lst):
...
-    no_of_elements = len(lst)
+    no_of_elements = len(collection)
     for _ in range(
         int(((no_of_elements - 1) / 2) + 1)
     ):  # we don't need to traverse to end of list as
         for j in range(no_of_elements - 1):
             if (
-                lst[j + 1] < lst[j]
+                collection[j + 1] < collection[j]
             ):  # applying bubble sort algorithm from left to right (or forwards)
```
Comment (cclauss) [inline sorts/double_sort.py:27]:
```suggestion
            ):
```

**Review by cclauss — APPROVED** [review]
Nice!!  Thanks for doing this.

**Comment by Krishna-Singhal** [comment]
> Nice!! Thanks for doing this.

Could you please add `hacktoberfest-accepted` label?

**Pull request merged**